cmd/livepeer_bench/livepeer_bench.go

```diff
@@ -0,0 +1,216 @@
+	// incorrectly add their own flags (specifically, due to the 'testing'
+	// package being linked)
+	flag.Set("logtostderr", "true")
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fname := flag.String("fname", "", "Input m3u8 manifest file")
+	conc := flag.Int("conc", 1, "# of concurrent transcode sessions")
+	segs := flag.Int("segs", 0, "Maximum # of segments to transcode (default all)")
+	profs := flag.String("profs", "P240p30fps16x9,P360p30fps16x9,P720p30fps16x9", "Transcoding options for broadcast job, or path to json config")
+	nvidia := flag.String("nvidia", "", "Comma-separated list of Nvidia GPU device IDs to use for transcoding")
+	outPrefix := flag.String("outPrefix", "", "Output segments' prefix (no segments are generated by default)")
+
+	flag.Parse()
+
+	if *fname == "" {
+		panic("Please provide the input manifest as `-fname <input.m3u8>`. See -h or -help for more.")
+	}
+
+	profiles := parseVideoProfiles(*profs)
+
+	f, err := os.Open(*fname)
+	if err != nil {
+		panic(err)
+	}
+	p, _, err := m3u8.DecodeFrom(bufio.NewReader(f), true)
+	if err != nil {
+		panic(err)
+	}
+	pl, ok := p.(*m3u8.MediaPlaylist)
+	if !ok {
+		panic("Expecting media PL")
+	}
+
+	accel := ffmpeg.Software
+	devices := []string{}
+	if *nvidia != "" {
+		accel = ffmpeg.Nvidia
+		devices = strings.Split(*nvidia, ",")
+	}
+
+	ffmpeg.InitFFmpeg()
+	segCount := 0
+	var wg sync.WaitGroup
+	dir := path.Dir(*fname)
+	start := time.Now()
+	fmt.Fprintf(os.Stderr, "Program %s Source %s Concurrency %d Profiles %s\n", os.Args[0], *fname, *conc, *profs)
+	fmt.Println("time,stream,segment,length")
+	for i := 0; i < *conc; i++ {
+		wg.Add(1)
+		go func(k int, wg *sync.WaitGroup) {
+			tc := ffmpeg.NewTranscoder()
+			for j, v := range pl.Segments {
+				if *segs > 0 && j >= *segs {
+					break
+				}
+				if v == nil {
+					continue
+				}
+				u := path.Join(dir, v.URI)
+				in := &ffmpeg.TranscodeOptionsIn{
+					Fname: u,
+					Accel: accel,
+				}
+				if ffmpeg.Software != accel {
+					in.Device = devices[k%len(devices)]
+				}
+				profs2opts := func(profs []ffmpeg.VideoProfile) []ffmpeg.TranscodeOptions {
+					opts := []ffmpeg.TranscodeOptions{}
+					for n, p := range profs {
+						oname := ""
+						if *outPrefix != "" {
+							oname = fmt.Sprintf("%s_%s_%d_%d_%d.ts", *outPrefix, p.Name, n, k, j)
+						} else {
+							oname = "-"
+						}
+						o := ffmpeg.TranscodeOptions{
+							Oname:        oname,
+							Profile:      p,
+							Accel:        accel,
+							AudioEncoder: ffmpeg.ComponentOptions{Name: "copy"},
+							Muxer:        ffmpeg.ComponentOptions{Name: "mpegts"},
+						}
+						opts = append(opts, o)
+					}
+					return opts
+				}
+				out := profs2opts(profiles)
+				t := time.Now()
+				_, err := tc.Transcode(in, out)
+				end := time.Now()
+				fmt.Printf("%s,%d,%d,%0.2v\n", end.Format("2006-01-02 15:04:05.999999999"), k, j, end.Sub(t).Seconds())
+				if err != nil {
+					panic(err)
```

Comment:
I don't think we want to panic within a goroutine this will stop the execution of the entire program but doesn't clean up running goroutines 

Should we opt to log the error (and possibly add this to the table per segment mentioned in my earlier comment) ? 

```go
if err != nil {
data = append(data, []string{
  fmt.Sprintf("%v", k),
  fmt.Sprintf("%v", j),
  fmt.Sprintf("%v s", time.Since(t).Seconds()),
  err.Error(), 
})
 glog.Error(err)
 continue
}
data = append(data, []string{
  fmt.Sprintf("%v", k),
  fmt.Sprintf("%v", j),
  fmt.Sprintf("%v s", time.Since(t).Seconds()),
  "n/a", 
})

```

Reply:
Ah, still learning the golang quirks. Will change :+1:
And table append sounds good.

Reply:
Changed to `glog.Fatalf` instead of `panic`

Decided to not go the *log-error-per-segment* route for now, as the transcoder session might not be in an OK state after throwing an error at us - so better to quit everything (different sessions are processing the same segments anyway)